pkg/provisioner/ironic/ironic.go

```diff
@@ -1867,3 +1868,40 @@ func (p *ironicProvisioner) loadBusyHosts() (hosts map[string]struct{}, err erro
 
 	return hosts, nil
 }
+
+func (p *ironicProvisioner) AddBMCEventSubscriptionForNode(subscription *metal3v1alpha1.BMCEventSubscription, httpHeaders provisioner.HTTPHeaders) (result provisioner.Result, err error) {
+	newSubscription, err := nodes.CreateSubscription(
+		p.client,
+		p.nodeID,
+		nodes.CallVendorPassthruOpts{
+			Method: "create_subscription",
+		},
+		nodes.CreateSubscriptionOpts{
+			Destination: subscription.Spec.Destination,
+			Context:     subscription.Spec.Context,
+			Protocol:    subscription.Spec.Protocol,
+			EventTypes:  subscription.Spec.EventTypes,
+			HttpHeaders: httpHeaders,
+		}).Extract()
+	if err != nil {
+		return provisioner.Result{RequeueAfter: subscriptionRequeueDelay}, err
```

Comment:
What's the point of requeueing on failure? At the very least we should check which error it does: it makes no sense to retry a subscription for a wrong driver.

Reply:
Retrying gives the user the chance to fix the subscription CR. We should definitely be logging the error though.

Reply:
I guess I'm worried about meaningless retries in case of e.g. unsupported driver (which is not possible to fix without recreating the BMH).